### DISCUSSION TITLE
Optimizers

### DISCUSSION THREAD

**Discussion by Mirat-pl-eu** [Smile-SA/elasticsuite#3325] [Q&A] [answered]
Please let me know why some attributes are available in the configuration and others are not?
What setting does it depend on?

**Comment by romainruaud** [comment]
To appear on the "Apply to products" section, an attribute must be at least "is_used_for_promo_rules=1", or "is_filterable=1", or "is_filterable_in_search=1".

If you have an attribute that you want to use for boosting but not display for layered navigation, go for the "is_used_for_promo_rules=1".

For the "Based on attribute values" boost, only the integer/decimal attributes are available (except Price).

Regards

**Comment by Mirat-pl-eu** [comment]
Thank you for your answer.
Do I understand correctly that the "text" attribute cannot be used?

**Reply by romainruaud** [reply]
They can be used in the "Apply to product" zone if they're set to be "used for promo rules".

**Comment by Mirat-pl-eu** [comment]
Can I use the quantity attribute? quantity_and_stock_status

I changed the settings, but it doesn't appear in the options to choose from, what could be?
![image](https://github.com/user-attachments/assets/c3cdc6d4-d987-4e94-bcea-5facc2c4b4e7)

**Comment by Mirat-pl-eu** [comment]
I want to strengthen the products we have in stock in the search.

**Reply by romainruaud** [reply, marked as answer]
Create a boost with a positive weight and then select the special attribute "Only in stock products" in the "Apply to Products" then.

Regards

**Reply by Mirat-pl-eu** [reply]
Thanks for the tip, but I had to use a different combination. For some reason, we have all products in stock.
